lib/ansiblelint/rules/__init__.py

```diff
@@ -190,7 +190,7 @@ def __init__(self, rulesdirs=None) -> None:
             self.extend(load_plugins(rulesdir))
         self.rules = sorted(self.rules, key=lambda r: r.id)
 
-    def register(self, obj):
+    def register(self, obj: AnsibleLintRule):
```

Comment:
How is this related to tests?

Reply:
Yes it does. I would really recommend you to try adding some typing information yourself on ansible-lint, because that is something you will realise right away: adding type to one method does create a long chain of dependencies.

Here is what happens if I remove it:
```
test/TestWithSkipTagId.py:10: error: Call to untyped function "register" in typed context
test/TestMetaMainHasInfo.py:61: error: Call to untyped function "register" in typed context
test/TestUseHandlerRatherThanWhenChanged.py:77: error: Call to untyped function "register" in typed context
test/TestComparisonToEmptyString.py:28: error: Call to untyped function "register" in typed context
test/TestSudoRule.py:48: error: Call to untyped function "register" in typed context
test/TestLineTooLong.py:17: error: Call to untyped function "register" in typed context
test/TestVariableHasSpaces.py:47: error: Call to untyped function "register" in typed context
test/TestMetaChangeFromDefault.py:19: error: Call to untyped function "register" in typed context
test/TestMetaVideoLinks.py:23: error: Call to untyped function "register" in typed context
test/TestTaskNoLocalAction.py:17: error: Call to untyped function "register" in typed context
test/TestDeprecatedModule.py:19: error: Call to untyped function "register" in typed context
test/TestShellWithoutPipefail.py:73: error: Call to untyped function "register" in typed context
test/TestEnvVarsInCommand.py:79: error: Call to untyped function "register" in typed context
test/TestRoleRelativePath.py:41: error: Call to untyped function "register" in typed context
test/TestMetaTagValid.py:17: error: Call to untyped function "register" in typed context
test/TestComparisonToLiteralBool.py:46: error: Call to untyped function "register" in typed context
Found 16 errors in 16 files (checked 117 source files)
```